src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -525,19 +527,41 @@ public static void TestRoundtrippingSpecificCodepageEncoding(string encodingName
         [MemberData(nameof(CodePageInfo))]
         public static void TestCodepageEncoding(int codePage, string webName, string queryString)
         {
-            Encoding encoding = CodePagesEncodingProvider.Instance.GetEncoding(queryString);
-            Assert.NotNull(encoding);
-            Assert.Equal(codePage, encoding.CodePage);
-            Assert.Equal(encoding, CodePagesEncodingProvider.Instance.GetEncoding(codePage));
-            Assert.Equal(webName, encoding.WebName);
-            Assert.Equal(encoding, CodePagesEncodingProvider.Instance.GetEncoding(webName));
-
-            // Small round-trip for ASCII alphanumeric range (some code pages use different punctuation!)
-            // Start with space.
-            string asciiPrintable = " 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
-
-            char[] traveled = encoding.GetChars(encoding.GetBytes(asciiPrintable));
-            Assert.Equal(asciiPrintable.ToCharArray(), traveled);
+            // There are two names that have duplicate associated CodePages. For those two names,
+            // we have to test with the expectation that querying the name will always return the
+            // same codepage.
+            if (codePage != 20932 && codePage != 50222)
+            {
+                Encoding encoding = CodePagesEncodingProvider.Instance.GetEncoding(queryString);
+                Assert.NotNull(encoding);
+                Assert.Equal(codePage, encoding.CodePage);
+                Assert.Equal(encoding, CodePagesEncodingProvider.Instance.GetEncoding(codePage));
```

Comment:
Is this line and the GetEncoding call above the only differences between the if and else blocks?  If so, can we just restrict those lines to being the duplicated ones, rather than duplicating everything?